src/python/pants/backend/python/sandboxed_interpreter.py

```diff
@@ -0,0 +1,64 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pex.interpreter import PythonInterpreter
+
+from pants.util.memo import memoized_method
+
+
+class SandboxedInterpreter(PythonInterpreter):
```

Comment:
Some pydocs would be good here.

Reply:
Agreed, especially given the discussion over whether this is the right way to do this.